_posts/2018-08-19-openfaas-on-fargate.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Faas on Fargate&#58; run OpenFaaS on AWS Fargate
+description: Edward Wilde introduces running OpenFaaS on AWS Fargate - a way to deploy and run functions without managing servers
+date: 2018-08-02
+image: /images/openfaas-on-fargate/astronomy-atmosphere-earth-220201.jpg
+categories:
+  - fargate
+author_staff_member: ed
+---
+## Why run OpenFaaS on AWS Fargate
+One of the great things about OpenFaaS is it's portability. You can run OpenFaaS out of the box using 
+officially supported platforms like are Kubernetes and Docker Swarm . The [community](https://github.com/openfaas/faas/blob/master/community.md) 
+has added other platforms that allow you to run OpenFaaS in more places. A good example of this is deploying OpenFaaS
+on Hashicorp's [nomad](https://www.hashicorp.com/blog/functions-as-a-service-with-nomad) application scheduler.
+
+### Complexity
+This brings us on to running OpenFaaS on AWS Fargate. Why would you want to do this? Largely this comes down to 
+complexity and cost. Perhaps you work in a team that has little or no knowledge of application schedulers like
+Kubernetes or Nomad. You may also work in a company that hasn't adopted Kubernetes and doesn't plan to in the immediate 
+future. So if you fit that in that camp, how do you leverage the advantage of serverless computing? A good choice would
+be to look at a managed serverless platform like [lambda](https://aws.amazon.com/lambda/). In this article we examine 
+another option, running OpenFaaS on Fargate, which I believe gives you some of the benefits of a managed service like 
+lambda with the flexibility of an opensource serverless platform like OpenFaaS.
+
+### Cost
+An interesting aspect of running a serverless platform, how much is it going to cost you? Working this out on
+paper we see that for small numbers of functions fargate is significantly cheaper:
+
+#### Fargate
+[AWS Fargate Pricing](https://aws.amazon.com/fargate/pricing/)
+
+| Region              |[us-east-1](https://docs.aws.amazon.com/general/latest/gr/rande.html) |
+|:------------------|:------------|
+| Container specification:  |   0.25 vCPU with 512MB          |
```

Comment:
Is 512MB enough for things like Prometheus?

Reply:
Good spot, i think default prom settings are tailored for 3GB memory allocation.

I will adjust the pricing accordingly 
Ref: https://prometheus.io/docs/prometheus/1.8/storage/#memory-usage

Reply:
i've removed the cost section